### ISSUE TITLE
BB keep firing into walls

### ISSUE THREAD

**Issue by Licho1** [ZeroK-RTS/Zero-K#2675]
Bb seems to ignore terrain and keeps firing through it.

https://i.imgur.com/9jWFRk2.jpg

**Comment by sprunk** [comment]
Bertha ignoring terrain is intentional because BB shots smooth terrain and can tear down terraform. Trying to tear down a whole mountain is bad though.

**Comment by GoogleFrog** [comment]
BB has to fire into walls to be usable because of UHM (unsynced heightmap). Without shooting into walls the following would occur:
 - Aim BB at the ground in enemy base.
 - BB shoots into enemy base, cratering is like to cause the ground height to increase at the position of your attack command.
 - BB is now aiming underground, it cannot shoot through ground. If on hold fire it will stop firing. If not on hold fire it will find to most useless target on the map and fire at it (in my experience).

Perhaps, you say, that the height of the attack command should be fit to the actual map height, not the height of the players UHM. This is problematic because the enemy base could have hidden spires or holes that cause your attack command to either be pointless or blocked by unseen terrain. There are still fairly common situations in which BB would need to shoot through terrain.

On another point, I feel like it is fine that BB can shoot through terrain because one of its uses is to flatten wall. It is nice to be able to tell your BB to shoot at a point, knocking down all walls in the way. If BB did not shoot through terrain you would need to target each successive wall manually and these walls may not even be scouted.

**Comment by Licho1** [comment]
In this case it was trying to shoot through fully visible wall of significant width.. but yeah it might be hard to calculate when it makes no sense...

**Comment by Licho1** [comment]
"Simple" solution would be a switch similar to sniper's "shoot on los only" ..
"shoot on clear path only"

**Comment by GoogleFrog** [comment]
Such a switch is possible. It would be a CallAsTeam to check the state of the UHM along its path. I think that you should use BB on manual fire though so such a switch would be rarely useful.
